test/Spec.hs

```diff
@@ -971,6 +967,61 @@ main =
       it "has no maintainer" $ ruleCatchesNot hasNoMaintainer "FROM debian"
       it "using add" $ ruleCatches copyInsteadAdd "ADD file /usr/src/app/"
 
+      it "has healthcheck" $
+        let dockerFile =
+              [ "FROM busybox"
+              ]
+        in ruleCatches hasHealthcheck $ Text.unlines dockerFile
+
+      it "not has healthcheck" $
+        let dockerFile =
+              [ "FROM busybox",
+                "HEALTHCHECK CMD stat /"
+              ]
+        in ruleCatchesNot hasHealthcheck $ Text.unlines dockerFile
+
+      it "multistage with healthcheck in all stages passes DL4007 and DL4008" $
+        let dockerFile =
+              [ "FROM debian:buster as stage1",
+                "HEALTHCHECK CMD check-1",
+                "FROM debian:buster",
+                "HEALTHCHECK CMD check-2"
+              ]
+        in do
+          ruleCatchesNot hasHealthcheck $ Text.unlines dockerFile
+          ruleCatchesNot multipleHealthcheck $ Text.unlines dockerFile
+
+      it "multistage with healthcheck in last stage only passes" $
+        let dockerFile =
+              [ "FROM debian:buster as stage1",
+                "FROM debian:buster",
+                "HEALTHCHECK CMD check-2"
+              ]
+        in ruleCatchesNot hasHealthcheck $ Text.unlines dockerFile
+
+      it "multistage with healthcheck in early stage only passes" $
+        let dockerFile =
+              [ "FROM debian:buster as stage1",
+                "HEALTHCHECK CMD check-1",
+                "FROM debian:buster"
+              ]
+        in ruleCatchesNot hasHealthcheck $ Text.unlines dockerFile
```

Comment:
I don't understand how this example passes the rule? The last `FROM` has no `HEALTCHECK` and is not inheriting from the first `FROM`

Reply:
That's not at all what this rule was supposed to check.
DL4007 was only about whether or not you have defined a `HEALTHCHECK` somewhere in you Dockerfile.

I have grown a few grey hairs over a revised version that has more sophisticated behaviour with multistage builds since.

Now this case and the case below are handled correctly and tested for - as well as the more complicated cases of multistage builds with and without consecutive build stages.